patches/sitecustomize.py

```diff
@@ -3,41 +3,50 @@
-        if explicit_project_id is None and specified_credentials is None and not has_bigquery:
-            print("Using Kaggle's public dataset BigQuery integration.")
-            return PublicBigqueryClient(*args, **kwargs)
+        # If this Kernel has bigquery integration on startup, preload the Kaggle Credentials
+        # object for magics to work. 
+        if get_integrations().has_bigquery():
+            from google.cloud.bigquery import magics
+            magics.context.credentials = KaggleKernelCredentials()
 
-        else:
-            if specified_credentials is None:
-                kwargs['credentials'] = KaggleKernelCredentials()
-                if (not has_bigquery):
-                    print('Please ensure you have selected a BigQuery '
-                          'account in the Kernels Settings sidebar.')
-            return bq_client(*args, **kwargs)
+        def monkeypatch_bq(bq_client, *args, **kwargs):
+            specified_credentials = kwargs.get('credentials')
+            has_bigquery = get_integrations().has_bigquery()
+            # Prioritize passed in project id, but if it is missing look for env var. 
+            arg_project = kwargs.get('project')
+            explicit_project_id = arg_project or os.environ.get(environment_vars.PROJECT)
+            # This is a hack to get around the bug in google-cloud library.
+            # Remove these two lines once this is resolved:
+            # https://github.com/googleapis/google-cloud-python/issues/8108
+            if explicit_project_id:
+                kwargs['project'] = explicit_project_id
+            if explicit_project_id is None and specified_credentials is None and not has_bigquery:
+                print("Using Kaggle's public dataset BigQuery integration.")
+                return PublicBigqueryClient(*args, **kwargs)
 
-    # Monkey patches BigQuery client creation to use proxy or user-connected GCP account.
-    # Deprecated in favor of Kaggle.DataProxyClient().
-    # TODO: Remove this once uses have migrated to that new interface.
-    bq_client = bigquery.Client
-    bigquery.Client = lambda *args, **kwargs:  monkeypatch_bq(
-        bq_client, *args, **kwargs)
+            else:
+                if specified_credentials is None:
+                    kwargs['credentials'] = KaggleKernelCredentials()
+                    if (not has_bigquery):
+                        print('Please ensure you have selected a BigQuery '
+                            'account in the Kernels Settings sidebar.')
+                return bq_client(*args, **kwargs)
+
+        # Monkey patches BigQuery client creation to use proxy or user-connected GCP account.
+        # Deprecated in favor of Kaggle.DataProxyClient().
+        # TODO: Remove this once uses have migrated to that new interface.
+        bq_client = bigquery.Client
+        bigquery.Client = lambda *args, **kwargs:  monkeypatch_bq(
+            bq_client, *args, **kwargs)
+
+init()
```

Comment:
Don't you want to call init only if the kernel has the integration?

Reply:
My idea here was "init the sitecustomize logic", basically leave this program unchanged in logic but just make it callable to reproduce starting up in tests. I wanted to be able to unit test if "KAGGLE_KERNEL_INTEGRATIONS" is set for a Kernel will the the monkeypatching logic behave correctly on startup.

Reply:
So, if I were to only call init if the kernel has the integration, then I wouldn't be to test that logic in the unit test.

Reply:
Makes sense. Thanks for the clarification.